test/fork/migration-v2.js

```diff
@@ -896,40 +894,18 @@ describe('V2 upgrade', function () {
     expect(tcBalanceDiff).to.be.equal(cnLockedAmountSum);
   });
 
-  // TODO review
-  // We should also read
-  // `const { amount, lastDistributionRound } = await ps.accumulatedRewards(coverable);`
-  // and check if the amount > 0
-  // Also, we should iterate through all products, in case we deprecated something that had a cover buy since the
-  // last reward distribution round
   it('Call LegacyPooledStaking.pushRewards for all non-deprecated contracts', async function () {
-    const PRODUCT_ADDRESSES_OUTPUT = require(PRODUCT_ADDRESSES_OUTPUT_PATH);
-    const productAddresses = PRODUCT_ADDRESSES_OUTPUT.map(address => address.toLowerCase());
-
-    /**
-     * pushRewards is affected by 2 other code flows:
-     * accumulateRewards (that itself calls pushRewards and accumulates rewards)
-     * pushBurn (that itself calls pushRewards)
-     *
-     * Post bulk upgrade to v2 contracts at the step:
-     * Deploy & upgrade contracts: MR, MCR, CO, TC, PS, PriceFeedOracle, P1, CL (CoverMigrator), GW, CR
-     *
-     * There is no other code path that can trigger accumulateRewards and pushBurn.
-     */
-
-    console.log(`Call pushRewards with ${productAddresses.length} product addresses.`);
-    await this.pooledStaking.pushRewards(productAddresses);
+    const productsWithPossibleRewards = require(PRODUCTS_WITH_REWARDS_PATH).map(address => address.toLowerCase());
```

Comment:
I would have simply relied on `contracts.json` (including deprecated products) instead

Reply:
I extracted the list using dune, to make sure we have absolutely all products that ever had rewards. I know we don't have to push all of them (for ex if there're 0 amounts) but I want the have it on-chain just in case. Less work for us and safer checks. Doesn't cost too much because it's called a single time.